packages/@webex/internal-plugin-metrics/src/call-diagnostic/call-diagnostic-metrics.ts

```diff
@@ -375,30 +380,48 @@ export default class CallDiagnosticMetrics extends StatelessWebexPlugin {
    * @param rawError
    */
   generateClientEventErrorPayload(rawError: any) {
+    if (rawError.name) {
+      if (isBrowserMediaErrorName(rawError.name)) {
+        return this.getErrorPayloadForClientErrorCode({
+          serviceErrorCode: undefined,
+          clientErrorCode: BROWSER_MEDIA_ERROR_NAME_TO_CLIENT_ERROR_CODES_MAP[rawError.name],
+        });
+      }
+    }
+
     const serviceErrorCode =
-      rawError?.error?.body?.errorCode || rawError?.body?.errorCode || rawError?.body?.code;
+      rawError?.error?.body?.errorCode ||
+      rawError?.body?.errorCode ||
+      rawError?.body?.code ||
+      rawError?.body?.reason?.reasonCode;
+
     if (serviceErrorCode) {
       const clientErrorCode = SERVICE_ERROR_CODES_TO_CLIENT_ERROR_CODES_MAP[serviceErrorCode];
       if (clientErrorCode) {
         return this.getErrorPayloadForClientErrorCode({clientErrorCode, serviceErrorCode});
       }
 
-      // by default, if it is locus error, return nre locus err
+      // by default, if it is locus error, return new locus err
       if (isLocusServiceErrorCode(serviceErrorCode)) {
         return this.getErrorPayloadForClientErrorCode({
           clientErrorCode: NEW_LOCUS_ERROR_CLIENT_CODE,
           serviceErrorCode,
         });
       }
+    }
 
-      // otherwise return meeting info
+    if (isMeetingInfoServiceError(rawError)) {
```

Comment:
in the code block above (lines 398-411) we check if the error is in `SERVICE_ERROR_CODES_TO_CLIENT_ERROR_CODES_MAP` and I see there are many error codes from WbxAppApi in that mapping, so I think if we get an error from WbxAppApi meeting info, we will execute the return from line 401, right? and we won't get to this code that seems to be written specifically to handle meeting info errors from WbxAppApi. Or am I misunderstanding something?

Reply:
You said 'I think if we get an error from WbxAppApi meeting info, we will execute the return from line 401'. Actually only if we get any error THAT is in our SERVICE_ERROR_CODES_TO_CLIENT_ERROR_CODES_MAP, then we return line 401. But if we get an error that is not in that map for any chance, at least we can detect if it is a Locus error (lines 404-411) or WbxAppApi error (lines 413-418)

Reply:
ok, yeah, I misunderstood the intention of this change, after speaking offline I understand that this code is added just as a fallback if `SERVICE_ERROR_CODES_TO_CLIENT_ERROR_CODES_MAP` mapping doesn't catch the code.